felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -2067,10 +2080,14 @@ func (m *bpfEndpointManager) getInterfaceIP(ifaceName string) (*net.IP, error) {
 }
 
 func (m *bpfEndpointManager) onServiceUpdate(update *proto.ServiceUpdate) {
-	if !m.ctlbWorkaroundEnabled {
+	if m.ctlbWorkaroundEnabled == ctlbWorkaroundDisabled {
 		return
 	}
 
+	// XXX It would be great to exclude other than UDP services if
+	// m.ctlbWorkaroundEnabled == ctlbWorkaroundUDPOnly but ServiceUpdate does
+	// not carry the protocol :(
```

Comment:
If it's in the k8s Service object you can add this trivially.  See dataplane_passthru.go.  It's literally just copying the fields we need from the k8s Service to our own protobuf struct (which you can add fields to by editing felixbackend.proto) :-)